balancer/weightedtarget/weightedtarget.go

```diff
@@ -143,6 +143,12 @@ func (b *weightedTargetBalancer) UpdateClientConnState(s balancer.ClientConnStat
 
 	b.targets = newConfig.Targets
 
+	// To send a TF state update in the case of a configuration with no targets
```

Comment:
Could you please add some more detail in this comment because a bunch of things are happening here:
- `len(b.targets) == 0` means that we have removed all child policies from the balancer group and the aggregator
- a call to `b.stateAggregator.ResumeStateUpdates` is deferred at the start of this method
- so, setting the `needUpdateStateOnResume` to `true` here will ensure that a new picker is built as part of the above deferred function, and since there are no child policies, the aggregated connectivity state in `Aggregator` will be `TransientFailure`.

Another thing to ensure here would be the error message returned by the picker in this case. If this is the logic that gets execute in this scenario to build a new picker: https://github.com/grpc/grpc-go/blob/master/balancer/weightedtarget/weightedaggregator/aggregator.go#L254-L261, the `weightedPickerGroup` will not have any child pickers and the `Next()` method would return `nil`, and we would end up with `ErrNoSubConnsAvailable`. See: https://github.com/grpc/grpc-go/blob/master/balancer/weightedtarget/weightedaggregator/aggregator.go#L296-L300. Maybe that is what we want, but it might be a good idea to check if C-core returns a more descriptive error.

Reply:
Done.